rocAL/rocAL/include/meta_data/coco_meta_data_reader.h

```diff
@@ -37,21 +38,32 @@ class COCOMetaDataReader: public MetaDataReader
     void release() override;
     void print_map_contents();
     bool set_timestamp_mode() override { return false; }
-
+    std::pair<uint32_t,uint32_t> get_max_size() override { return std::make_pair(_max_height,_max_width); }
     const std::map<std::string, std::shared_ptr<MetaData>> & get_map_content() override { return _map_content;}
     COCOMetaDataReader();
 private:
     pMetaDataBatch _output;
     std::string _path;
     int meta_data_reader_type;
     void add(std::string image_name, BoundingBoxCords bbox, Labels labels, ImgSize image_size, uint image_id = 0);
-    void add(std::string image_name, BoundingBoxCords bbox, Labels labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count); // To add Mask coordinates to Metadata struct
+    void add(std::string image_name, BoundingBoxCords bbox, Labels labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count, MetaDataType meta_data_type); // To add Mask coordinates to Metadata struct
```

Comment:
Why add an extra param? Use _output->get_metadata_type() inside the function to get the type and do ops accordingly